src/Pages/Recipe/Recipe.js

```diff
@@ -0,0 +1,167 @@
+import "./Recipe.css";
+import NutritionModal from "../../components/NutritionModal/NutritionModal";
+import { MdOutlineCircle } from "react-icons/md";
+import DOMPurify from "dompurify";
+
+const Recipe = () => {
+  const [data, setData] = useState("");
+  const [ingredients, setIngredients] = useState([]);
+  const [instructions, setInstructions] = useState([]);
+  const [favorite, setFavorite] = useState(true);
+  const [isFav, setIsFav] = useState("");
+  const [err, setErr] = useState("");
+
+  let params = useParams();
+  const url = `/api/v1/recipes/${params.id}`;
+  const token = localStorage.getItem("myKitchenAppToken");
+
+  const fetchRecipe = async () => {
+    const recipe = await axios.get(url, {
+      headers: { Authorization: "Bearer " + token },
+    });
+    return recipe;
+  };
+
+  useEffect(() => {
+    fetchRecipe()
+      .then((response) => {
+        setData(response.data);
+        setIngredients(response.data.ingredients);
+        setInstructions(response.data.instructions);
+        setIsFav(response.data.isFavorite);
+      })
+      .catch((error) => console.log(error));
+  }, []);
+
+  const add = async () => {
+    const favRecipe = {
+      recipeId: data.id,
+      userId: "",
+      title: data.title,
+      image: data.image,
+    };
+
+    try {
+      const { recipe } = await axios.post("/api/v1/recipes/list/", favRecipe, {
+        headers: {
+          Authorization: "Bearer " + token,
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+      });
+      setFavorite(favRecipe);
+      setIsFav(true);
+    } catch (err) {
+      setErr(err.message);
+    }
+  };
+
+  const remove = async () => {
+    try {
+      const { recipe } = await axios.delete(
+        `/api/v1/recipes/${params.id}`,
+
+        {
+          headers: {
+            Authorization: "Bearer " + token,
+            "Content-Type": "application/json",
+            Accept: "application/json",
+          },
+        }
+      );
+      setIsFav(false);
+    } catch (err) {
+      setErr(err.message);
+    }
+  };
+
+  const sanitizedData = () => ({
+    __html: DOMPurify.sanitize(data.summary),
+  });
+
+  return (
+    <>
+      <div className="pageWrapper">
+        <div className="topContainer">
+          <img alt={data.title} src={data.image} />
+          <div className="recipeInfo">
+            <h1>{data.title}</h1>
+            <ul>
+              <li>
+                <ImSpoonKnife className="icon" />
+                {data.servings} servings.
+              </li>
+              <li>
+                <HiClock className="icon" />
+                ready in {data.readyInMinutes} mins.
+              </li>
+            </ul>
+            <div>
+              {isFav ? (
+                <Button
+                  color="error"
+                  type="submit"
+                  size="large"
+                  onClick={remove}
+                >
+                  <FavoriteIcon />
```

Comment:
It would be good if an explanation came out when I pressed the heart.
<img width="216" alt="Screen Shot 2022-11-27 at 9 22 54 PM" src="https://user-images.githubusercontent.com/88150284/204200213-e84e8173-6cc2-457a-b3e8-87850fa69ae6.png">

Reply:
I added the text "Saved" here.